### PR TITLE
Fix build on macos

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note

```

### CODE REVIEW

File: proxy_components/proxy_ffi/src/jemalloc_utils.rs [hunk 1]
```diff
@@ -63,7 +63,28 @@ pub fn issue_mallctl_args(
...
                 // Happens only with `raftstore-proxy-main`
-                return mallctl(c_ptr, oldptr, oldsize, newptr, newsize);
+                #[cfg(not(any(test, feature = "testexport")))]
+                {
+                    // No test part
+                    #[cfg(feature = "external-jemalloc")]
+                    {
+                        // Must linked to tiflash.
+                        return mallctl(c_ptr, oldptr, oldsize, newptr, newsize);
+                    }
```
Comment (JaySon-Huang) [inline proxy_components/proxy_ffi/src/jemalloc_utils.rs:73]: Is it conflict with line 63?

File: proxy_components/proxy_ffi/src/jemalloc_utils.rs [hunk 2]
```diff
@@ -63,7 +63,28 @@ pub fn issue_mallctl_args(
             #[cfg(not(feature = "external-jemalloc"))]
             {
                 // Happens only with `raftstore-proxy-main`
-                return mallctl(c_ptr, oldptr, oldsize, newptr, newsize);
+                #[cfg(not(any(test, feature = "testexport")))]
+                {
```
Comment (JaySon-Huang) [inline proxy_components/proxy_ffi/src/jemalloc_utils.rs:67]: Is it duplicated with line 55?

File: proxy_components/proxy_ffi/src/jemalloc_utils.rs (hunk 1 above)
Comment (CalvinNeo) [reply proxy_components/proxy_ffi/src/jemalloc_utils.rs:73]: Fixed

File: proxy_components/proxy_ffi/src/jemalloc_utils.rs (hunk 2 above)
Comment (CalvinNeo) [reply proxy_components/proxy_ffi/src/jemalloc_utils.rs:67]: Fixed

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**